gemrb/core/Scriptable/Actor.cpp

```diff
@@ -5400,15 +5416,19 @@ bool Actor::CheckOnDeath()
 	return false;
 }
 
-void Actor::IncrementDeathVariable(Variables *vars, const char *format, StringView name) const {
+void Actor::IncrementDeathVariable(Game::kaputz_t& vars, const char *format, StringView name) const {
 	if (!name.empty()) {
 		ieVariable varname;
 		if (!varname.Format(format, name)) {
 			Log(ERROR, "Actor", "Scriptname {} (name: {}) is too long for generating death globals!", name, fmt::WideToChar{GetName()});
 		}
-		ieDword start = 0;
-		vars->Lookup(varname, start);
-		vars->SetAt(varname, start + 1, nocreate);
+
+		auto lookup = vars.find(varname);
+		if (lookup != vars.cend()) {
+			lookup->second += 1;
+		} else if (!nocreate) {
+			vars[varname] = 1;
+		}
```

Comment:
We now have 4 blocks like this in the same file, so it's ripe for a static function.